Prevent Elastic-licensed code from leaking into OSS distributions

The Beats projects releases OSS-only artifacts (e.g. [filebeat OSS-only](https://www.elastic.co/downloads/beats/filebeat-oss)). These artifacts should not contain code or features that have an Elastic license (aka "x-pack" code for legacy reasons). The project should have enforcement that there are no dependencies from OSS code to x-pack code.

This can be enforced with [depguard](https://github.com/OpenPeeDeeP/depguard) check. For example:

```yaml
# .golangci-lint.yml
linters:
  enable:
    - depguard
linters-settings:
  depguard:
    rules:
      apache-licensed-code:
        list-mode: lax
        files:
          - '!**/x-pack/**/*.go'
        deny:
          - pkg: github.com/elastic/beats/v7/x-pack
            desc: Apache 2.0 licensed code cannot depend on Elastic licensed code (x-pack/).
```

The current state is that there is a single violation:

```
filebeat/input/filestream/internal/input-logfile/harvester_test.go:36:2: import 'github.com/elastic/beats/v7/x-pack/dockerlogbeat/pipelinemock' is not allowed from list 'apache-licensed-code': Apache 2.0 licensed code cannot depend on Elastic licensed code (x-pack/). (depguard)
        "github.com/elastic/beats/v7/x-pack/dockerlogbeat/pipelinemock"
        ^

```

References

- https://github.com/elastic/beats/blob/0959d568d633e38cfa117cd1acf04da78f62424f/LICENSE.txt

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -66,9 +66,19 @@ linters:
     - wastedassign # wastedassign finds wasted assignment statements.
     - gomodguard # check for blocked dependencies
     - prealloc # Finds slice declarations that could potentially be pre-allocated
+    - depguard
 
 # all available settings of specific linters
 linters-settings:
+  depguard:
+    rules:
+      apache-licensed-code:
+        list-mode: lax
+        files:
+          - '!**/x-pack/**/*.go'
+        deny:
+          - pkg: github.com/elastic/beats/v7/x-pack
+            desc: Apache 2.0 licensed code cannot depend on Elastic licensed code (x-pack/).
   errcheck:
     # report about not checking of errors in type assertions: `a := b.(MyStruct)`;
     check-type-assertions: false

